### PR TITLE
blockdev_commit_fio: update timeout for fio run and commit job

### PR DESCRIPTION
**Pull request by aliang123**
Update timeout for fio.run and commit job

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:2039134

### CODE REVIEW

File: qemu/tests/blockdev_commit_fio.py [hunk 1]
```diff
@@ -47,6 +51,8 @@ def run_test(self):
             self.verify_data_file()
             bg_test.join()
```
Comment (yanan-fu) [inline qemu/tests/blockdev_commit_fio.py:52]: No need to join(), is it ?
As our discussion offline, `time_base` can guarantee fio is running when commit snapshot start (time_base > sleep_max),  it is not must to make fio running until `time_base` reached.

File: qemu/tests/blockdev_commit_fio.py [hunk 2]
```diff
@@ -15,11 +15,12 @@ def fio_thread(self):
         fio_options = self.params.get("fio_options")
         if fio_options:
             logging.info("Start to run fio")
-            fio = generate_instance(self.params, self.main_vm, 'fio')
+            self.fio = generate_instance(self.params, self.main_vm, 'fio')
+            fio_run_timeout = int(self.params.get("fio_timeout", 2400))
```
Comment (yanan-fu) [inline qemu/tests/blockdev_commit_fio.py:19]: @aliang123  Let's use `params.get_numeric`

File: qemu/tests/blockdev_commit_fio.py [hunk 3]
```diff
@@ -32,7 +33,10 @@ def commit_snapshots(self):
             cmd, args = commit_cmd(device)
             job_id = args.get("job-id", device)
             self.main_vm.monitor.cmd(cmd, args)
-            job_utils.wait_until_block_job_completed(self.main_vm, job_id, timeout=1800)
+            job_timeout = int(self.params.get("commit_job_timeout", 1800))
```
Comment (yanan-fu) [inline qemu/tests/blockdev_commit_fio.py:36]: Same,  `params.get_numeric`

File: qemu/tests/cfg/blockdev_commit_fio.cfg [hunk 4]
```diff
@@ -18,5 +18,7 @@
     mount_point = "/var/tmp"
     sleep_min = 300
     sleep_max = 480
+    fio_timeout = 2400
+    commit_job_timeout = 1800
     fio_options = '--name=stress --filename=/home/atest --ioengine=libaio --rw=write --direct=1 '
-    fio_options += '--bs=4K --size=2G --iodepth=256 --numjobs=256 --runtime=1800'
+    fio_options += '--bs=4K --size=1G --iodepth=256 --numjobs=256 --time_based=${fio_timeout}'
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_commit_fio.cfg:24]: --time_based --runtime=${fio_timeout}

**Comment by yanan-fu** [comment]
@aliang123  @ngu-niny Could you provide the test result for different arch ? Thanks!

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_commit_fio.q35: PASS (1311.82 s)

**Comment by yanan-fu** [comment]
@zhencliu  Do you have any more comments ? Thanks!

**Comment by ngu-niny** [comment]
With the timeout values, the test on ppc would fail. I am still debugging after some modification.

**Comment by zhencliu** [comment]
> @zhencliu Do you have any more comments ? Thanks!
@aliang123 @ngu-niny 

Let's wait Nini's update later, if it's hard to use the same timeout value, what about defining it this way
ppc64, ppc64le:
    timeout = xxx
x86_64:
    timeout = yyy 
So we can assign different timeout values  for both ppc and x86.

**Comment by yanan-fu** [comment]
+1  depends on the test result for different arch to adjust the timeout value accordingly.

**Comment by ngu-niny** [comment]
@aliang123  From the latest test result, 'commit_job_timeout = 1800' is not enough, while the value 2400 works, so can we set 'commit_job_timeout = 2400' only for ppc64le?

**Comment by aliang123** [comment]
> @aliang123 From the latest test result, 'commit_job_timeout = 1800' is not enough, while the value 2400 works, so can we set 'commit_job_timeout = 2400' only for ppc64le?

@ngu-niny, have updated, please help to check again on ppc64le, thanks.

**Comment by ngu-niny** [comment]
@aliang123 Please also add s390x for the same timeout setting as ppc64le. Thank you.

**Comment by ngu-niny** [comment]
(1/1) repeat1.Host_RHEL.m8.u6.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.blockdev_commit_fio:  PASS (2579.28 s)

(1/1) repeat1.Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.s390x.io-github-autotest-qemu.blockdev_commit_fio:  PASS (2543.71 s)

Acked-by: Nini Gu <ngu@redhat.com>

**Comment by zhencliu** [comment]
hi @ngu-niny do we need to have a test against s390x?

**Comment by zhencliu** [comment]
As tests passed on x86/ppc/s390x.

LGTM

**Review by yanan-fu — APPROVED** [review]
Ack

**Pull request merged**